SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+        if (this.head[0] == apple[0] && this.head[1] == apple[1]) {
+            tiles[apple[0]][apple[1]].classList.remove("apple");
+            this.appleCount++;
+            this.game.apple = game.newAppleLoc();
+            pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+        } else {
+            tail = this.squeue.dequeue().data;
+        }
+
+        game.tick(this.head, tail);
+    }
+}
+
+class Game {
+    constructor(n) {
+        this.n = n;
+    }
+
+    reset() {
+        this.board = [];
+        for (let i = 0; i < this.n; i++) {
+            let row = [];
+            for (let j = 0; j < this.n; j++) {
+                row.push('-');
+            }
+            this.board.push(row);
+        }
+
+        this.snake = new Snake(this);
+        this.gameState = "NOT_STARTED";
+
+        // reset tiles on page
+        let allTiles = document.getElementsByClassName("snakeBody");
+        while (allTiles.length > 0) {
+            allTiles[0].classList.remove("snakeBody");
+        }
+        let appleTiles = document.getElementsByClassName("apple");
+        while (appleTiles.length > 0) {
+            appleTiles[0].classList.remove("apple");
+        }
+
+        this.apple = this.newAppleLoc();
+        this.debug();
+
+        this.interval = setInterval(function () {
+            game.snake.move();
+        }, 200);
+
+        pageMessages.innerHTML = "Game State: " + game.gameState;
+    }
+
+    newAppleLoc() {
+        let loc = [Math.round(Math.random() * (this.n - 1)), Math.round(Math.random() * (this.n - 1))];
+        if (this.board[loc[0]][loc[1]] != "-") {
+            return this.newAppleLoc();
+        }
+        console.log("new apple location: " + loc);
+
+        // change text board
+        this.board[loc[0]][loc[1]] = "@";
+
+        // change html
+        tiles[loc[0]][loc[1]].classList.add("apple");
+
+        return loc;
+    }
+
+    debug() {
+        console.log(this.board.join("\n"));
+    }
+
+    end() {
+        clearInterval(this.interval);
+        console.log("Apples collected: " + this.snake.appleCount);
+        this.gameState = "ENDED";
+        pageMessages.innerHTML = "Game State: " + this.gameState;
+    }
+
+    tick(head, tail) {
+
+        // check if 
+        if (head[0] >= this.n || head[1] >= this.n || head[0] < 0 || head[1] < 0) {
+            this.end();
+            return;
+        }
+        console.log(head + " " + tail);
+
+        if (this.board[head[0]][head[1]] == '$') {
+            this.end();
+            return;
+        }
+        // html page board colors
+        tiles[head[0]][head[1]].classList.add("snakeBody");
+
+        if (tail != null) {
+            tiles[tail[0]][tail[1]].classList.remove("snakeBody");
+        }
+
+        // text board
+        this.board[head[0]][head[1]] = '$';
+        if (tail != null) {
+            this.board[tail[0]][tail[1]] = '-';
+        }
+        this.debug();
+    }
+}
+
+let boardSize = 12;
```

Comment:
You may want to move all of this top-level code into a function (e.g., something like `init()`), otherwise it feels a bit scattered. But this is okay.